var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -255,6 +255,8 @@ class Trilinos(CMakePackage):
     patch('xlf_seacas.patch', when='@12.10.1:%xl_r')
     patch('xlf_tpetra.patch', when='@12.12.1:%xl')
     patch('xlf_tpetra.patch', when='@12.12.1:%xl_r')
+    patch('xlf_seacas.patch', when='@12.12.1:%clang')
```

Comment:
Hopefully does not break clang+gfortran

Reply:
@davydden As far as the `xlf_seacas.patch` is concerned, it only affects code applied when cmake identifies the compiler as IBM XL fortran (`ELSEIF ("${CMAKE_Fortran_COMPILER_ID}" MATCHES "XL")`).  While the `xlc_seacas.patch` is applied over code that will be run in the clang+fortran case,  I've been able to verify that I can still build `trilinos` for clang+gfortran with this pull request applied.